_pytest/junitxml.py

```diff
@@ -71,6 +71,8 @@ def __init__(self, nodeid, xml):
         self.nodes = []
         self.testcase = None
         self.attrs = {}
+        self.testsuite_properties = {}
```

Comment:
I think we should follow the same semantics as used by `self.properties`: keep a single list instead of a dict and a list to preserve order. This way we even allow duplicate properties, which are perfectly valid.